packages/wonder-blocks-data/util/types.js

```diff
@@ -1,4 +1,24 @@
 // @flow
+export type CacheEntry<TData> =
+    | {|
+          error: string,
+          data?: ?void,
+      |}
+    | {|
+          data: TData,
+          error?: ?void,
+      |};
```

Comment:
This type is a little confusing.  For disjoint unions it's better to have a discriminator field, something like:
```
type CacheEntry<TData> =
  | {| type: "data", data: TData |}
  | {| type: "error", error: string |}
```

Reply:
Prettier did that, not me.